### PR TITLE
Change to Plan project

### PR DESCRIPTION
**Pull request by michaelmarshall**
#1738 
https://github.com/SeedCompany/cord-api-v3/issues/1788

### CODE REVIEW

File: src/components/project/project.resolver.ts [hunk 1]
```diff
@@ -58,6 +58,15 @@ class ModifyOtherLocationArgs {
   locationId: ID;
 }
 
+@ArgsType()
+class ReadProjectArgs {
+  @IdField()
+  id: ID;
+
+  @IdField({ nullable: true })
+  changeId?: ID;
```
Comment (zkhin) [inline src/components/project/project.resolver.ts:67]: Aware CR read request.

File: src/components/project/project.service.ts [hunk 2]
```diff
@@ -554,6 +555,20 @@ export class ProjectService {
       pinned: !!result.pinnedRel,
       scope: result.memberRoles.flat().map(rolesForScope('project')),
     };
+
+    if (changeId) {
```
Comment (zkhin) [inline src/components/project/project.service.ts:559]: I was planning to return only CR pending properties and other fields should be undefined or null but several fields are non-nullable in Project DTO. So I returned origin active properties if they are not CR requested.

File: src/components/project/project.service.ts [hunk 3]
```diff
@@ -626,8 +641,13 @@ export class ProjectService {
           : InternshipProject,
       object: currentProject,
       changes: simpleChanges,
+      changeId,
```
Comment (zkhin) [inline src/components/project/project.service.ts:644]: Refactor updateProperties with `changeId` field.

File: src/core/database/database.service.ts [hunk 4]
```diff
@@ -226,25 +226,39 @@ export class DatabaseService {
...
         continue;
       }
-      await this.updateProperty({
-        type,
-        object,
-        key: prop as any,
-        value: change,
-      });
+      if (changeId) {
+        await this.updatePlanChangeProperty({
```
Comment (zkhin) [inline src/core/database/database.service.ts:246]: Defined updating plan change property.
```
(:Project)-[:name {active: false}]->(:Property)<-[:change {active: true}]-(:PlanChange)
```

File: src/components/project/change-to-plan/plan-change.service.ts [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    { filter, ...input }: ChangeListInput,
+    session: Session
+  ): Promise<ChangeListOutput> {
+    // const label = 'PlanChange';
+
+    const query = this.db
+      .query()
+      .match([
+        // requestingUser(session),
+        // ...permissionsOfNode(label),
```
Comment (zkhin) [inline src/components/project/change-to-plan/plan-change.service.ts:216]: TODO

File: src/components/project/project.service.ts (hunk 2 above)
Comment (CarsonF) [reply src/components/project/project.service.ts]: Yeah that's what I was thinking anyways 👍

**Comment by zkhin** [comment]
`PlanChange` label is used for querying/storing `status` , `summary` and `types` props. 
```
(Project)-[:changeset]->(PlanChange)->[:status]->(status:Property)
```
`Changeset` label is used for querying/storing pending project changes. 
```
(Project)-[:changeset]->(Changeset)->[:name]-(name:Property)
```

**Review by CarsonF — APPROVED** [review]
🎉 🎉 🎉 

tenor-167373723

**Pull request merged**